pkg/distcache/provider.go

```diff
@@ -173,27 +236,69 @@ func (constructor DistCacheConstructor) ProvideDistCache(in DistCacheConstructor
 
 	dc.Olric = o
 
+	dc.jobGroup = in.JobGroup
+	job := &jobs.BasicJob{
+		JobBase: jobs.JobBase{
+			JobName: distCacheMetricsJobName,
+		},
+		JobFunc: dc.scrapeMetrics,
+	}
+	jobConfig := jobs.JobConfig{
+		InitialDelay:     config.MakeDuration(0),
+		ExecutionPeriod:  config.MakeDuration(time.Millisecond * 500),
+		ExecutionTimeout: config.MakeDuration(time.Millisecond * 1000),
+		InitiallyHealthy: false,
+	}
+	dc.Metrics = newDistCacheMetrics()
+
 	in.Lifecycle.Append(fx.Hook{
 		OnStart: func(ctx context.Context) error {
+			// Register metrics with Prometheus.
+			err := dc.Metrics.registerMetrics(in.PrometheusRegistry)
+			if err != nil {
+				log.Error().Err(err).Msg("Failed to register distcache metrics with Prometheus registry")
+				return err
+			}
+
 			panichandler.Go(func() {
 				err = dc.Olric.Start()
+				if err != nil {
+					log.Error().Err(err).Msg("Failed to start distcache")
+				}
 				_ = in.Shutdowner.Shutdown()
 			})
 			// wait for olric to start by waiting on startChan until ctx is canceled
 			select {
 			case <-ctx.Done():
 				return errors.New("olric failed to start")
 			case <-startChan:
+				err := dc.jobGroup.RegisterJob(job, jobConfig)
+				if err != nil {
+					log.Error().Err(err).Msg("Failed to register distcache scrape metrics job with jobGroup")
+					return err
+				}
 			}
 			return nil
 		},
 		OnStop: func(ctx context.Context) error {
-			var multiErr error
+			// Unregister metrics with Prometheus.
+			err := dc.Metrics.unregisterMetrics(in.PrometheusRegistry)
```

Comment:
these operations should be in reverse order of how they start - which means this should be called the last.